### ISSUE TITLE
Update store loop exited while executing the tests

### ISSUE THREAD

**Issue by Kerollmops** [meilisearch/MeiliSearch#1619]
**Describe the bug**

```
---- index::delete_index::loop_delete_add_documents stdout ----
thread 'tokio-runtime-worker' panicked at 'Update store loop exited', meilisearch-http/src/index_controller/update_actor/store/mod.rs:251:13
thread 'index::delete_index::loop_delete_add_documents' panicked at 'assertion failed: `(left == right)`
  left: `500`,
 right: `202`: {"message":"Internal error: panic","errorCode":"internal","errorType":"internal_error","errorLink":"https://docs.meilisearch.com/errors#internal"}', meilisearch-http/tests/index/delete_index.rs:36:9
```

**To Reproduce**
1. Go to the MeiliSearch repository
2. Just do a `cargo test`

**MeiliSearch version:** v0.21.0, even just main.

**Additional context**
I am on macos on an Apple M1 machine.

**Comment by MarinPostma** [comment]
I am working on a fix for this.

**Comment by curquiza** [comment]
Thanks Marin, this bug fix is not expected for v0.22.0 anyway but for v0.23.0, so take your time

**Comment by curquiza** [comment]
Need a redesign of the update store to solve this issue https://github.com/meilisearch/MeiliSearch/issues/1679
I remove it from v0.23.0 Milestones

**Comment by curquiza** [comment]
Should be solved in v0.25.0 then!
I did not succeed to reproduce.
@Kerollmops can you check again on your machine (launching cargo test 3-4 times)?

**Comment by Kerollmops** [comment]
Seems to be fixed on the latest commit of the main branch ea0a5271.
